### ISSUE TITLE
[BUG] viewOnceMessage

### ISSUE THREAD

**Issue by mrfzvx12** [WhiskeySockets/Baileys#1151]
i get this output when trying to detect viewonce message, message is not read.

{
  key: {
    remoteJid: '*****@s.whatsapp.net',
    fromMe: true,
    id: '60D61E2BAA5D9A207A80A82A87C808EF',
    participant: undefined
  },
  messageTimestamp: 1732793287,
  pushName: "Kleo",
  broadcast: false,
  status: 2,
  messageStubType: 2,
  messageStubParameters: [ 'Message absent from node']
}

**Comment by AtaliasOliveira1** [comment]
I have the same problem!

**Comment by Salientekill** [comment]
I heard that it was due to a WhatsApp update that was causing this problem, but in the latest versions it has already been fixed.

**Comment by bifurx** [comment]
I have the same problem too :(

**Comment by mrfzvx12** [comment]
> I heard that it was due to a WhatsApp update that was causing this problem, but in the latest versions it has already been fixed.

I am using the latest version but the result is still the same, sometimes it works but then has the same problem again.

**Comment by Sansekai** [comment]
Same problem

**Comment by pxawtyy** [comment]
same

**Comment by CSFelix** [comment]
Same!!

**Comment by Akane7101** [comment]
![Screenshot_٢٠٢٤١٢٠٨-١٧٠٤٥٩_WhatsApp](https://github.com/user-attachments/assets/5e3170c7-a3f6-4e97-84d0-a8fb808b57f7)
it won't be detected yes..but you can still decrypt it

**Comment by mrfzvx12** [comment]
> ![Screenshot_٢٠٢٤١٢٠٨-١٧٠٤٥٩_WhatsApp](https://private-user-images.githubusercontent.com/129458531/393615629-5e3170c7-a3f6-4e97-84d0-a8fb808b57f7.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3MDEzNjcsIm5iZiI6MTczMzcwMTA2NywicGF0aCI6Ii8xMjk0NTg1MzEvMzkzNjE1NjI5LTVlMzE3MGM3LWEzZjYtNGU5Ny04NGQwLWE4ZmI4MDhiNTdmNy5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIwOFQyMzM3NDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZDY2NzNjOGQyZjlhMDVkZmZiZDllODdkOTBhNDM0NGY2NmNkZDExOWNjNDAzYWNiYjQyMmIwMmUwOGRkOWNiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.yQVuJ0E7JBCv4TSX5u9LLQXwI_CCPZHOFRaqveYgh1Q) it won't be detected yes..but you can still decrypt it

yes it still works if we reply to viewonce message but it doesn't work if we directly detect the message.

**Comment by Loki-Xer** [comment]
Same issue i tryed Bailey last version it 🐢 not fixed

**Comment by purpshell** [comment]
Message wouldn't be absent from node on view once, unless something has changed in the structure of <message> node recently. Worth investigating

**Comment by polvora6969** [comment]
Same problem 12/25/2024

**Comment by polvora6969** [comment]
I comment that the fault is directly with Android operating systems, iOS does detect them correctly.
Whatsapp changed some structure only on droid

**Comment by diasEduardo** [comment]
> > ![Screenshot_٢٠٢٤١٢٠٨-١٧٠٤٥٩_WhatsApp](https://private-user-images.githubusercontent.com/129458531/393615629-5e3170c7-a3f6-4e97-84d0-a8fb808b57f7.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3MDEzNjcsIm5iZiI6MTczMzcwMTA2NywicGF0aCI6Ii8xMjk0NTg1MzEvMzkzNjE1NjI5LTVlMzE3MGM3LWEzZjYtNGU5Ny04NGQwLWE4ZmI4MDhiNTdmNy5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIwOFQyMzM3NDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZDY2NzNjOGQyZjlhMDVkZmZiZDllODdkOTBhNDM0NGY2NmNkZDExOWNjNDAzYWNiYjQyMmIwMmUwOGRkOWNiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.yQVuJ0E7JBCv4TSX5u9LLQXwI_CCPZHOFRaqveYgh1Q) it won't be detected yes..but you can still decrypt it
> 
> yes it still works if we reply to viewonce message but it doesn't work if we directly detect the message.

you are right, got nothing from the message but  when I reply it came at

console.log(msg.message?.extendedTextMessage?.contextInfo?.quotedMessage?.viewOnceMessage)
console.log(msg.message?.extendedTextMessage?.contextInfo?.quotedMessage?.viewOnceMessageV2?.message)

**Comment by Epikcoder** [comment]
> ![Screenshot_٢٠٢٤١٢٠٨-١٧٠٤٥٩_WhatsApp](https://private-user-images.githubusercontent.com/129458531/393615629-5e3170c7-a3f6-4e97-84d0-a8fb808b57f7.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY3MzI1ODEsIm5iZiI6MTczNjczMjI4MSwicGF0aCI6Ii8xMjk0NTg1MzEvMzkzNjE1NjI5LTVlMzE3MGM3LWEzZjYtNGU5Ny04NGQwLWE4ZmI4MDhiNTdmNy5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDExM1QwMTM4MDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05YTg5OGI1YmJhMzQzMzNmNWM3ZWE1YzdkNzQzNjhlZDY3MzI5YThkZGNiODdmZDVhZjA3ODEzY2QwMjc2ODQ0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.i9uWUH0FYOmDGYQWFtVNxh9QKANgccxwuQ1QQrXmF_w) it won't be detected yes..but you can still decrypt it

IIRC, This only works on mobile devices,
trying to quote the media using whatsapp desktop/web would result on empty media

**Comment by AstroX11** [comment]
The issue occurs mostly on WhatsApp desktop

**Comment by CSFelix** [comment]
I've just tested with Baileys latest version (`6.7.12`) and faced the same issue. When sending/receiving view once messages, the following message object is returned regardless of the message being an image, audio or video:

```js
{
  key: {
    remoteJid: <jid>,
    fromMe: true,
    id: <message_id>,
    participant: undefined
  },
  messageTimestamp: 1738583767,
  pushName: <pushname>,
  broadcast: false,
  status: 3,
  messageStubType: 2,
  messageStubParameters: [ 'Message absent from node' ]
}
```

For some reason, view once messages are being `absent from node`.

**Comment by canove** [comment]
Not a bug, changes in whatsapp caused it, closing.

**Comment by tguimmaraess** [comment]
> Not a bug, changes in whatsapp caused it, closing.

Whats the solution?
